app/controllers/versions_controller.rb

```diff
@@ -11,4 +12,25 @@ def show
     @versioned_links = @rubygem.links(@latest_version)
     render "rubygems/show"
   end
+
+  def dependencies_show
+    params[:version_id] ? latest_version_by_slug : latest_version
+    @dependencies = @latest_version.dependencies
+    render "transitive_dependencies/show"
+  end
+
+  def transitive_dependencies_show
+    latest_version
+    @gem_details = @latest_version.dependencies
+    @deps_names = @gem_details.collect { |dependency| dependency.rubygem.name }
+    respond_to do |format|
+      format.js { render "transitive_dependencies/show" }
```

Comment:
Hey, lets change this to return JSON response, move the ` show.js.erb` code to `assets/javascript`. It is better to stick with the asset pipeline than compiling JS on request.